### DISCUSSION TITLE
Conditional support in jsonschema generator?

### DISCUSSION THREAD

**Discussion by mogul** [linkml/linkml#988] [Q&A] [answered]
Hi folks,

We're in the middle of a spike to evaluate LinkML for potential use in a federal project at [GSA TTS](https://www.gsa.gov/about-us/organization/federal-acquisition-service/technology-transformation-services).

There's signs of conditional rule support having been added in 1.2 and coming to jsconschema. For example, [the last paragraph of this section of the docs](https://linkml.io/linkml/schemas/advanced.html?highlight=conditional#unions-as-ranges) (possibly stray since 1.3 is out now?) says:

> In the 1.2 series:
> - generated python should use a Union
> - jsonschema should use [conditionals](https://json-schema.org/understanding-json-schema/reference/conditionals.html)
> - OWL should use UnionOf

...and there's mention of `adding conditionals` in [this big refactoring commit](https://github.com/linkml/linkml/commit/770f7c0421de9b986f0d35a9e05780bcd57b6a8f).

However, when we use the jsonschema generator, we see no sign of them coming through, and we can't find any indication that anyone is working on this in a PR anywhere, or other generator examples to look at.

What's the best way to follow, support, and possibly contribute to having conditional rules make it all the way through to generated JSON Schema?

Thanks,
Bret

**Comment by sierra-moxon** [comment]
Thank you Bret!  I appreciate you taking the time to look through the documentation and release notes/code.  You're right, I can't find a specific issue for this either; I created [one](https://github.com/linkml/linkml/issues/989).   I'll add the 4 upvotes here to that case ([feature voting](https://github.com/linkml/linkml/issues/974)).  We welcome any and all contributions in the form of tickets, PRs, comments, etc. :)

**Reply by mogul** [reply]
We saw the issue get created; thanks so much! In particular, we're after the `preconditions` and `postconditions` [described in the doc](https://linkml.io/linkml/schemas/advanced.html?highlight=conditional#rules)... Would those be in-scope for that issue?

**Reply by cmungall** [reply, marked as answer]
The issue was resolved last year jsonschemagen has full support, thanks!
